buildpacks/ruby/tests/integration_test.rs

```diff
@@ -112,6 +111,51 @@ fn test_ruby_app_with_yarn_app() {
         );
 }
 
+#[test]
+#[ignore = "integration test"]
+fn test_barnes_app() {
+    TestRunner::default().build(
+        BuildConfig::new("heroku/builder:22", "tests/fixtures/barnes_app"),
+        |context| {
+            assert_contains!(context.pack_stdout, "# Heroku Ruby Buildpack");
+
+            context.start_container(
+                ContainerConfig::new()
+                    .entrypoint("launcher")
+                    .envs(vec![
+                        ("DYNO", "web.1"),
+                        ("PORT", "1234"),
+                        ("AGENTMON_DEBUG", "1"),
+                        ("HEROKU_METRICS_URL", "example.com"),
+                    ])
+                    .command(["while true; do sleep 1; done"]),
```

Comment:
> I fixed the tests (https://github.com/heroku/buildpacks-ruby/pull/172#issuecomment-1714857517) to be more deterministic, but it's still based on a timeout (which can fail if the machine is slow or docker is slow due to timing conditions). It's better than it was, though. Please take a look. Specifically, this line does not seem great https://github.com/heroku/buildpacks-ruby/pull/172/files#diff-17c88161984214fafac5afdc89a5919f2dbcfb1165f2560fd255809316095fe8R131.

The `while started.elapsed() < Duration::from_secs(20)` timeout seems fine to me -- it's no different from some of the other timeouts we have in both buildpack implementations or integration tests IMO (eg download timeouts).

The `while true; do sleep 1; done` is fine too - though searching now I just learned that GNU `sleep` supports an `infinity` option too:
https://unix.stackexchange.com/questions/366077/is-there-a-linux-command-that-does-nothing-but-never-exits

That said, if you really didn't like the sleep, perhaps the testcase could be updated to have a `Procfile` that actually starts Puma, and then the testcase also has the advantage of exercising the "can barnes actually report to the statsd server successfully" part too. (It's fine if you want to skip this for now and handle later.)

Reply:
I like `sleep infinity`. Functionally it's about the same, but it seems more intentional. However when I tried it I started to get failures (even after bumping my max time to 30 seconds), so it might not be interpreted as the same thing. 

```
thread 'test_barnes_app' panicked at 'assertion failed: `(left contains right)`
left (unescaped):
/layers/heroku_ruby/metrics_agent/agentmon_loop already running.
Logging enabled via `AGENTMON_DEBUG="1"`. To disable `unset AGENTMON_DEBUG`

left (escaped): `"/layers/heroku_ruby/metrics_agent/agentmon_loop already running.\nLogging enabled via `AGENTMON_DEBUG=\"1\"`. To disable `unset AGENTMON_DEBUG`"`
right: `"Booting agentmon_loop"`', buildpacks/ruby/tests/integration_test.rs:152:21
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

We might want to call out: "Here's how to boot a container that will run until your block exits" in the docs for a use case like this.

Reply:
> We might want to call out: "Here's how to boot a container that will run until your block exits" in the docs for a use case like this.

I'm not sure that's really necessary, since:
- most libcnb-test integration tests should either (a) be running one-off commands using `run_shell_command` (as recommended already in the docs), or (b) if they need to be running servers, then they should actually run the servers not fake processes (ideally the tests here would be running Puma as mentioned above)
- there's a tipping point for docs-length over which people stop reading or miss important details because they are hidden in-between less important information, and so we should try not to add general programming/operating system knowledge in here unless it's going to affect lots of users and also hard to figure out (in this case the first result for `"linux command that never exits"` gives the exact answer)